shared/middleware/channels_api.rb

```diff
@@ -164,7 +164,9 @@ class ChannelsApi < Sinatra::Base
   # Marks the specified channel as published.
   #
   post %r{/v3/channels/([^/]+)/publish/([^/]+)} do |channel_id, project_type|
-    not_authorized if under_13?
+    not_authorized unless owns_channel?(channel_id)
+    not_authorized unless %w(artist playlab applab gamelab).include?(project_type)
+    not_authorized if under_13? && !%w(artist playlab).include?(project_type)
```

Comment:
In the `owns_channel?` and `under_13?` cases the better response would be `forbidden` (User is authenticated but not allowed to access resource) rather than `not_authorized` (User is not authenticated).  It would be reasonable to add a fourth case (listed first) that returned `not_authorized` if the user is not signed in.  I think failing the project_type whitelist should probably return `bad_request`.

Here's [a stackoverflow post on 401 NOT AUTHORIZED vs 403 FORBIDDEN](https://stackoverflow.com/questions/3297048/403-forbidden-vs-401-unauthorized-http-responses), and [a pretty awesome infographic on server request handling and response types](https://www.loggly.com/blog/http-status-code-diagram/) from Loggly.